src/mattack_actors.cpp

```diff
@@ -291,58 +297,46 @@ mattack_actor *gun_actor::clone() const
     return new gun_actor( *this );
 }
 
-// @todo Enforce load order (add finalization?) and allow calculating this at json load
-int gun_actor::get_range() const
+bool gun_actor::call( monster &z ) const
 {
-    if( target_range > 0 ) {
-        return target_range;
-    }
+    Creature *target;
 
-    item gun( gun_type );
-    itype_id ammo = ( ammo_type != "NULL" ) ? ammo_type : gun.ammo_default();
-
-    if( ammo != "NULL" ) {
-        gun.ammo_set( ammo, max_ammo );
-    }
-
-    return gun.gun_range( true );
-}
+    if( z.friendly ) {
+        int max_range = 0;
+        for( const auto& e : ranges ) {
+            max_range = std::max( { max_range, e.first.first, e.first.second } );
+        }
 
-bool gun_actor::call( monster &z ) const
-{
-    const int range = get_range();
-    if( z.friendly != 0 ) {
-        // Attacking monsters, not the player!
-        int boo_hoo;
-        Creature *hostile_target = z.auto_find_hostile_target( range, boo_hoo );
-        if( hostile_target == nullptr ) {
-            // Couldn't find any targets!
-            if( boo_hoo > 0 && g->u.sees( z ) ) {
-                // because that stupid oaf was in the way!
+        int obstacles; // friendly targets blocking shots againt hostiles
```

Comment:
The comment is wrong. The number is the number of enemies who can't be attacked because attacking them would risk attacking the player.